### PR TITLE
Fix cmake dll copy destination

### PR DESCRIPTION
**Pull request by chrisdembia**
### Brief summary of changes

This PR fixes two small bugs that I've wanted to fix for a while:

1. AppVeyor used to give an error that it couldn't create a certain folder (`Release`, in the build dir), and I just figured out that this was caused by a small bug in our CMakeLists.
2. Before, if you built a single OpenSim test, this would not cause the Simbody libraries to get copied; you'd have to build *all* targets first. Now, the Simbody libraries get copied even if you are building a single test.

### Testing I've completed

AppVeyor.

### CHANGELOG.md (choose one)

- no need to update because...these changes are very minor.

The Doxygen for this PR can be viewed at http://myosin.sourceforge.net/?C=N;O=D; click the folder whose name is this PR's number.

### CODE REVIEW

**Comment by chrisdembia** [comment]
If possible, it would be great if we could merge this quickly. I'm using it in the direct collocation code. The appveyor tests passed on the second-to-last commit. The only tests that didn't pass were Travis Debug, which is unrelated.

**Comment by tkuchida** [comment]
> it would be great if we could merge this quickly

Looks innocuous enough to me, though I haven't tested. I'm satisfied if AppVeyor and Travis are satisfied.

File: OpenSim/Common/CMakeLists.txt [hunk 1]
```diff
@@ -21,6 +21,10 @@ OpenSimAddLibrary(
     )
 
 if(WIN32)
-    add_dependencies(osimCommon Simbody_CONFIG_check)
+    add_dependencies(osimCommon
+        Simbody_CONFIG_check Copy_Simbody_DLLs)
```
Comment (aseth1) [inline OpenSim/Common/CMakeLists.txt]: Maybe just add a comment why Simbody_CONFIG_check is Windows only. Thanks!

Comment (chrisdembia) [reply OpenSim/Common/CMakeLists.txt]: Done.

**Comment by chrisdembia** [comment]
Thanks @aseth1 and @tkuchida . Ready for review again.

**Comment by chrisdembia** [comment]
Don't merge this yet, I think there's an issue.

**Comment by chrisdembia** [comment]
I pushed a commit to fix an issue I just found. Waiting for CI tests...

**Comment by chrisdembia** [comment]
Okay this is ready for review again.

**Comment by chrisdembia** [comment]
Okay nevermind again; I have to fix yet another issue with this PR. Thanks @carmichaelong

**Comment by chrisdembia** [comment]
Alright. I fixed yet another issue in this PR and I think it is ready for review again!

**Comment by chrisdembia** [comment]
Thanks @carmichaelong; I'm merging under rule 5.

**Pull request merged**